community/kernel/src/main/java/org/neo4j/kernel/impl/proc/Procedures.java

```diff
@@ -136,10 +136,22 @@ public void registerProcedure( Class<?> proc ) throws KernelException
     /**
      * Register a new procedure defined with annotations on a java class.
      * @param proc the procedure class
+     * @param overrideCurrentImplementation set to true if procedures within this class should override older procedures with the same name
      */
     public void registerProcedure( Class<?> proc, boolean overrideCurrentImplementation ) throws KernelException
     {
-        for ( CallableProcedure procedure : compiler.compileProcedure( proc ) )
+        registerProcedure( proc, overrideCurrentImplementation, "" );
```

Comment:
I'd prefer not to give empty string special meaning. Better to use optional everywhere.